src/main/java/uk/ac/ebi/atlas/download/DownloadController.java

```diff
@@ -0,0 +1,66 @@
+package uk.ac.ebi.atlas.download;
+
+import org.apache.commons.net.ftp.FTPFile;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.RequestMapping;
+import uk.ac.ebi.atlas.controllers.HtmlExceptionHandlingController;
+
+import org.apache.commons.net.ftp.FTPClient;
+
+import java.io.IOException;
+import java.time.LocalDate;
+import java.time.ZoneId;
+import java.util.HashMap;
+import java.util.Map;
+
+@Controller
+public class DownloadController extends HtmlExceptionHandlingController {
+
+    private static final String filePath = "/pub/databases/microarray/data/atlas/experiments/atlas-latest-data.tar.gz";
+
+    @RequestMapping(value = "/download", produces = "text/html;charset=UTF-8")
+    public String getExperimentsListParameters(Model model) {
+
+        Map<String, String> fileInfo = getFTPFileInfo();
+
+        model.addAttribute("fileSize", fileInfo.get("fileSize"));
+        model.addAttribute("fileName", fileInfo.get("fileName"));
+        model.addAttribute("fileTimestamp", fileInfo.get("fileTimestamp"));
+
+        model.addAttribute("mainTitle", "Download ");
+
+        return "download";
+    }
+
+    private Map<String, String> getFTPFileInfo() {
+        Map<String, String> fileInfo = new HashMap<>();
+        FTPClient ftpClient = new FTPClient();
+        try {
+            ftpClient.connect("ftp.ebi.ac.uk");
+            ftpClient.login("ftp", "anonymous");
+            FTPFile[] file = ftpClient.listFiles(filePath);
+            fileInfo.put("fileName", file[0].getName());
+            fileInfo.put("fileSize", format(file[0].getSize(), 1));
+            fileInfo.put("fileTimestamp", LocalDate.ofInstant(
+                    file[0].getTimestamp().getTime().toInstant(),
+                    ZoneId.systemDefault()).toString());
+
+        } catch (IOException ex) {
+            ex.printStackTrace();
```

Comment:
If there is an error in the file then I think the download page should be displayed with missing data in the download file table instead of throwing a runtime exception and not displaying the page at all.
From UX perspective I'm a little confused regarding what will make more sense either the missing data or throwing an exception. In my personal opinion, I prefer having an empty field in the file table rather than throwing an exception.

Reply:
If you throw a `RuntimeException`, the logic in `HtmlExceptionHandlingController` or `JsonExceptionHandlingController` will catch it and display an error page with a message, with a `4xx` or `5xx` return code. That should be the solution when nothing better can be done.

So the first question you should ask yourself is, what should happen if the file cannot be found?
I can think of a few alternatives:
1. Don’t display the table with the file link at all.
2. Show something like *File not available* or similar in the file info fields.

What’s displayed in your view when the tarball can’t be found?

Reply:
Okay, I understand. In current implementation if the file cannot be found then everything is displayed as is except the table column for the file name, size and date are empty.
I prefer the first option of not showing the table if the file is not found.